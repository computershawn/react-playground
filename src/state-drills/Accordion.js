import React from 'react'

class Accordion extends React.Component {
    static defaultProps = {
        sections: [],
    }

    constructor(props) {
        super(props)
        this.state = { currentAccordion: 0 }
    }

    handleButtonClick(index) {
        this.setState({currentAccordion: index});
    }

    renderButtons() {
        return this.props.sections.map((item, index) => (
            <li key={index}>
                <button onClick={() => this.handleButtonClick(index)}>{item.title}</button>
                {this.state.currentAccordion === index && this.renderContent()}
            </li>
        ))
    }

    renderContent() {
        const current = this.state.currentAccordion;
        const theContent = this.props.sections[current].content;
        return <p>{theContent}</p>
    }

    render() {
        return (
            <ul>
                {this.renderButtons()}
            </ul>
        )
    }
}

export default Accordion;