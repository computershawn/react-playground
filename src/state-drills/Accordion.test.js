import React from 'react';
import ReactDOM from 'react-dom';
import Accordion from './Accordion';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

/* now showing all the content of the file */

describe(`Accordion Component`, () => {

    const sections = [
        {
            title: 'Section 1',
            content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
        },
        {
            title: 'Section 2',
            content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
        },
        {
            title: 'Section 3',
            content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
        }
    ]

    it('renders without errors', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Accordion sections={sections} />, div);
        ReactDOM.unmountComponentAtNode(div);
    })
    
    it('renders empty given no accordions', () => {
        const wrapper = shallow(<Accordion />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('closes the first tab and opens any clicked tab', () => {
        const wrapper = shallow(<Accordion sections={sections} />)
        wrapper.find('button').at(2).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('closes the first tab and opens any clicked tab', () => {
        const wrapper = shallow(<Accordion sections={sections} />)
        wrapper.find('button').at(1).simulate('click')
        wrapper.find('button').at(2).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})