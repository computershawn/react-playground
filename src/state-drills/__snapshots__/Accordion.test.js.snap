// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component closes the first tab and opens any clicked tab 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
    <p>
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
  </li>
</ul>
`;

exports[`Accordion Component closes the first tab and opens any clicked tab 2`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
    <p>
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
  </li>
</ul>
`;

exports[`Accordion Component renders empty given no accordions 1`] = `<ul />`;
